src/erc20FeeProxy.ts

```diff
@@ -18,7 +18,7 @@ export function createPaymentForFeeProxy(
   payment.block = event.block.number.toI32();
   payment.timestamp = event.block.timestamp.toI32();
   payment.txHash = event.transaction.hash;
-  payment.gasUsed = event.transaction.gasUsed;
+  payment.gasUsed = event.transaction.gasLimit;
```

Comment:
could you explain this change? gasLimit and gasUsed are not the same thing...

Reply:
@benjlevesque should be done on another PR, not related with Near so I'll remove it now.

But the gasLimit is different from the gasUsed. Example of a tx where we retrieve the wrong number:
https://etherscan.io/tx/0xd0876ba91f2ef8c0d4687d782bb785d1e8f7478ccee485e1bd59df4acf9143f3

```
{
        "id": "0x1702404698caebd7a13b736fa7255d94dfaea53ccc866e143b7bf5228755aa4a",
        "contractAddress": "0x370de27fdb7d1ff1e1baa7d11c5820a324cf623c",
        "tokenAddress": "0xb64ef51c888972c908cfacf59b47c1afbc0ab8ac",
        "to": "0x8400b234e7b113686bd584af9b1041e5a233e754",
        "txHash": "0xd0876ba91f2ef8c0d4687d782bb785d1e8f7478ccee485e1bd59df4acf9143f3",
        "gasUsed": "100000"
      },
```

Reply:
It's a typing issue from `graph-ts` that I quick-fixed with `gasLimit`

Reply:
The graph does not expost receipt information like gasUsed at the moment so we mistakenly store gasLimit instead. So the type "fix" is actually mandatory to support Near, it does not change the behaviour, only makes the error more explicit. I'll add `FIXME` and refer to the issue below, ok for you @benjlevesque  ?

The Graph Issue here: https://github.com/graphprotocol/graph-node/issues/3218
Similar discussion here: https://github.com/graphprotocol/graph-node/issues/2619

Example here: 
0xd0876ba91f2ef8c0d4687d782bb785d1e8f7478ccee485e1bd59df4acf9143f3
gasLimit = 100k, gasUsage = 71k

And from our subgraph:
```
{
        "id": "0x1702404698caebd7a13b736fa7255d94dfaea53ccc866e143b7bf5228755aa4a",
        "contractAddress": "0x370de27fdb7d1ff1e1baa7d11c5820a324cf623c",
        "tokenAddress": "0xb64ef51c888972c908cfacf59b47c1afbc0ab8ac",
        "to": "0x8400b234e7b113686bd584af9b1041e5a233e754",
        "txHash": "0xd0876ba91f2ef8c0d4687d782bb785d1e8f7478ccee485e1bd59df4acf9143f3",
        "gasUsed": "100000"
      },
```

Reply:
I suggest we store 0 instead?

Reply:
We can also do that yes but `gasLimit` is closer to the true answer so it depends on the usage we have. I'll check the impact and let you know.

Reply:
it's not used because of the issue you mentionned at the moment

Reply:
OK I just set it to 0